website/docs/api/plugins/plugin-sitemap.mdx

```diff
@@ -44,11 +44,23 @@ Accepted fields:
 | `priority` | `number \| null` | `0.5` | See [sitemap docs](https://www.sitemaps.org/protocol.html#xmlTagDefinitions) |
 | `ignorePatterns` | `string[]` | `[]` | A list of glob patterns; matching route paths will be filtered from the sitemap. Note that you may need to include the base URL in here. |
 | `filename` | `string` | `sitemap.xml` | The path to the created sitemap file, relative to the output directory. Useful if you have two plugin instances outputting two files. |
+| `createSitemapItems` | <code>[CreateSitemapItemsFn](#CreateSitemapItemsFn) \| undefined</code> | `undefined` | An optional function which can be used to transform and / or filter the items in the sitemap. |
 
 ```mdx-code-block
 </APITable>
 ```
 
+### Types {#types}
+
+#### `CreateSitemapItemsFn` {#CreateSitemapItemsFn}
+
+```ts
+type CreateSitemapItemsFn = (params: {
+  siteConfig: DocusaurusConfig;
+  routes: RouteConfig[];
+}) => Promise<SitemapItem[]>;
```

Comment:
```suggestion
type CreateSitemapItemsFn = (params: {
  siteConfig: DocusaurusConfig;
  routes: RouteConfig[];
  defaultCreateSitemapItems: CreateSitemapItemsFn:
}) => Promise<SitemapItem[]>;
```

Similar to https://docusaurus.io/docs/api/plugins/@docusaurus/plugin-content-blog#CreateFeedItemsFn

Note: the types are not 100% accurate but I think at least it's simple for users to understand. 
(by that, I mean, you do not need to call `defaultCreateSitemapItems` with a `defaultCreateSitemapItems` param)